### ISSUE TITLE
Cross-Validation for Early Stopping

### ISSUE THREAD

**Issue by michael-GitHub-0** [keras-team/keras#11400]
Early-stopping seems to suffer from high variance. The performance of the early-stopped model is often very different on unseen data, and is usually overfit if one uses the model at the best epoch and underfit if one uses the model at the last epoch. 

Would it make sense to use a cross-validated early-stopping metric at each epoch to estimate the performance of the model at that epoch on unseen data? I.e., take the model at epoch i; then use 9 / 10 of the data to move (gradient descent in NN) to epoch i + 1 ; repeat for all folds; record the average performance; use all of the training data to move to epoch i + 1; stop training when the recorded performance stops improving.
